beginner_source/dist_overview.rst

```diff
@@ -15,107 +15,62 @@ to the technology that can best serve your use case.
 Introduction
 ------------
 
-As of PyTorch v1.6.0, features in ``torch.distributed`` can be categorized into
-three main components:
-
-* `Distributed Data-Parallel Training <https://pytorch.org/docs/stable/generated/torch.nn.parallel.DistributedDataParallel.html>`__
-  (DDP) is a widely adopted single-program multiple-data training paradigm. With
-  DDP, the model is replicated on every process, and every model replica will be
-  fed with a different set of input data samples. DDP takes care of gradient
-  communication to keep model replicas synchronized and overlaps it with the
-  gradient computations to speed up training.
-* `RPC-Based Distributed Training <https://pytorch.org/docs/stable/rpc.html>`__
-  (RPC) supports general training structures that cannot fit into
-  data-parallel training such as distributed pipeline parallelism, parameter
-  server paradigm, and combinations of DDP with other training paradigms. It
-  helps manage remote object lifetime and extends the
-  `autograd engine <https://pytorch.org/docs/stable/autograd.html>`__ beyond
-  machine boundaries.
-* `Collective Communication <https://pytorch.org/docs/stable/distributed.html>`__
-  (c10d) library supports sending tensors across processes within a group. It
+The PyTorch Distributed library includes a collective of parallelism modules,
+a communications layer (C10D), and infrastructure for launching and
+debugging large training jobs.
+
+These Parallelism Modules offer high-level functionality and compose with existing models:
+* `Distributed Data-Parallel (DDP) <https://pytorch.org/docs/stable/generated/torch.nn.parallel.DistributedDataParallel.html>`__
+* `Fully Sharded Data-Parallel Training (FSDP) <https://pytorch.org/docs/stable/fsdp.html>`__
+* TODO what to mention about FSDP2 vs FSDP
+* `Tensor Parallel <https://pytorch.org/docs/stable/distributed.tensor.parallel.html>`__
+* TODO link to pipeline parallel after stable docs link exists
+
+DTensor and DeviceMesh are primitives used to build parallelism in terms of sharded or replicated tensors on N-dimensional process groups.
+* `DTensor <https://github.com/pytorch/pytorch/blob/main/torch/distributed/_tensor/README.md>`__ represents a tensor that is sharded and/or replicated, and communicates automatically to reshard tensors as needed by operations.
+* `DeviceMesh <https://pytorch.org/docs/stable/distributed.html#devicemesh>`__ abstracts multiple dimensions of parallelism and manages the underlying ProcessGroups used for collective communications,
+
+
+`C10D <https://pytorch.org/docs/stable/distributed.html>`__
```

Comment:
do we want to still keep it as "Collective communication (c10d) module"? "C10D" might be a familiar term for us but might not for the audience

Reply:
well, any proposal for what to name it then?

Reply:
This whole block looks a bit [weird](https://docs-preview.pytorch.org/pytorch/tutorials/2913/beginner/dist_overview.html) when rendered:
Move `offers both  collective communication APIs (e.g.,` to the same line? Also agree that c10d might be confusing, suggesting to spell it out and give abbreviation in parethesis:

`Collective communication (C10D) module <https://pytorch.org/docs/stable/distributed.html>`__ offers....

Reply:
Thanks! I decided to reformat this section a bit- added 'headings' for each subsection, and also renamed the c10d link.  lmk what you think about the new version.